src/SuperDumpService/Services/SuperDumpRepository.cs

```diff
@@ -172,6 +177,24 @@ public class SuperDumpRepository {
 			}
 		}
 
+		private bool SetHashAndCheckIfDuplicated(string bundleId, FileInfo archiveFile) {
+			if (!archiveFile.Exists) {
+				Console.WriteLine($"Unable to find file {archiveFile.FullName}! Aborting analysis.");
+				return false;
+			}
+			string md5 = Utility.Md5ForFile(archiveFile);
+			var duplicates = bundleRepo.GetAll().Where(b => b.Status != BundleStatus.Duplication && b.FileHash == md5);
+			if (duplicates.Any()) {
+				string originalBundleId = duplicates.First().BundleId;
```

Comment:
There _should_ only be one because other duplicates are filtered out. If there are more, it doesn't matter which one we choose.